File: 06 - Subprogramas/valor_referência.c
#include <stdio.h>

void por_valor(int x) {
    x += 10;
    printf("Dentro da função (x): %d\n", x);
}

void por_referencia(int *x) {
    *x += 10;
    printf("Dentro da função (*x): %d\n", *x);
}

int main() {
    int a = 5;
    printf("Antes da função (a): %d\n", a);
    por_valor(a);
    printf("Depois da função (a): %d\n", a);

    por_referencia(&a);
    printf("Depois da função com ponteiro (a): %d\n", a);

    return 0;
}

File: 06 - Subprogramas/valor_referência.js
function porValor(x) {
    x += 10;
    console.log("Dentro da função (x):", x);
}

function porReferencia(arr) {
    arr.push(99);
    console.log("Dentro da função (arr):", arr);
}

let a = 5;
console.log("Antes da função (a):", a);
porValor(a);
console.log("Depois da função (a):", a);

let b = [1, 2, 3];
console.log("Antes da função (b):", b);
porReferencia(b);
console.log("Depois da função (b):", b);
